jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunSettingsEditorPanel.java

```diff
@@ -62,7 +64,14 @@ public SamRunSettingsEditorPanel(Project project, TextCompletionProvider handler
         useTemplate.addActionListener(e -> updateComponents());
         addQuickSelect(templateFile.getTextField(), useTemplate, this::updateComponents);
         templateFile.addActionListener(new TemplateFileBrowseListener());
+
         updateComponents();
+        startRefreshResourceMappings();
+    }
+
+    private void startRefreshResourceMappings() {
+        ResourceMappingImpl.promiseMapping(project)
```

Comment:
The pattern we've used in the past is `ApplicationManager`'s `invokeLater` and `executeOnPooledThread` in order to push something into the background and then come back to update the UI. Happy to take a steer on what's better - but is there a reason for this approach over the other?